benchmarks/timeJava8/src/main/java/streams/Java8Streams.java

```diff
@@ -0,0 +1,204 @@
+        long ret = LongStream.of(v_P)
+                .filter(x -> (x & 0x15) != 0x11)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x15) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x15) == 0x10)
+                .reduce(0, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void flatMap_take_primitive(Blackhole bh) {
+        long ret = LongStream.of(v_P)
+                .flatMap(x -> LongStream.of(vLo_P)
+                        .map(dP -> dP * x))
+                .limit(shortRangingFactor)
+                .reduce(0, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void sum_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .reduce(0L, (a, b) -> a + b);
+
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void sumOfSquares_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .map(d -> d * d)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void sumOfSquaresEven_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .filter(x -> x % 2 == 0)
+                .map(x -> x * x)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void cart_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .flatMap(d -> Stream.of(vLo_B)
+                        .map(dP -> dP * d))
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void maps_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .map(x -> x + (x & 0xD) + 0xCAFED00D)
+                .map(x -> x + (x & 0xE) + 0xD15EA5E)
+                .map(x -> x + (x & 0xA) + 0xDABBAD00)
+                .map(x -> x + (x & 0xD) + 0xDEADBAAD)
+                .map(x -> x + (x & 0xB) + 0xDEADDEAD)
+                .map(x -> x + (x & 0xE) + 0xDEADFA11)
+                .map(x -> x + (x & 0xE) + 0xFFBADD11)
+                .map(x -> x + (x & 0xF) + 0x4B1D)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
+    }
+
+    @Benchmark
+    public void filters_boxed(Blackhole bh) {
+        Long ret = Stream.of(v_B)
+                .filter(x -> (x & 0x15) != 0x11)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x15) != 0x10)
+                .filter(x -> (x & 0x10) == 0x10)
+                .filter(x -> (x & 0x30) != 0x10)
+                .filter(x -> (x & 0x15) == 0x10)
+                .reduce(0L, (a, b) -> a + b);
+        bh.consume(ret);
```

Comment:
@DarkDimius - This seems like overkill.  Value speculation is fine as long as it's representative of what can be accomplished in real-world scenarios.  Because of the variation in the input list, the benchmark can't use value speculation to avoid actually finding the correct answer.

The thing to avoid is making the different tests too dependent on each other so that fusion doesn't just avoid unnecessary intermediates but actually shows that the entire exercise is pointless; and to avoid adding steps that aren't measured because all the work is done in the first step.  As you wrote it here, the first step is effectively a Bloom filter: you reject all values less than 237, and thereafter succeed on only 1/64th of them.

It's better to choose some set of simple algorithms that behave predictably, like ((x*8783 >> 7) ^ x) & 0x3 != 0 and (x*1144 >> 7) & 0x3 != 0 and x & 0x3 != 0.